### PR TITLE
Use absolute links in remark toc

### PR DESCRIPTION
**Pull request by Swizec**
This makes it so Remark table of contents use absolute links.

Use case is creating overall TOC for the whole page where individual links aren't just on the same page as their text.

Example where I use this and links don't work correctly with current remark-transform: https://swizec.github.io/intro-to-react-workshop/ (scroll down to setlist)

**Need help:** I want to make links prefix-paths aware for obvious reasons, but the `__PREFIX_PATHS__` global isn't set in `gatsby-remark-transform/src/extend-node-type.js`. I can't figure out why.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Ah nice! Thanks for wading in with a fix.

The pathPrefix is passed as a named argument to the extension API. See for example it being used here https://github.com/gatsbyjs/gatsby/blob/3960d0954cb236459915971b7ca581994ad8d52b/packages/gatsby-plugin-netlify/src/gatsby-node.js#L32

It's either an empty string if the user didn't run build with the `--prefix` flag or the prefix with it.

https://github.com/gatsbyjs/gatsby/blob/320a454e041055d57070a0f853da1089f6b1876b/packages/gatsby/src/utils/api-runner-node.js#L61

**Comment by Swizec** [comment]
Thanks @KyleAMathews that worked like a charm. I updated the code.

File: packages/gatsby-transformer-remark/src/extend-node-type.js [hunk 1]
```diff
@@ -178,8 +178,27 @@ module.exports = (
       } else {
         const ast = await getAST(markdownNode)
         const tocAst = mdastToToc(ast)
+
         let toc
         if (tocAst.map) {
+          const addSlugToUrl = function(node) {
+            if (node.url) {
+              node.url = [
+                store.getState().config.pathPrefix,
```
Comment (KyleAMathews) [inline packages/gatsby-transformer-remark/src/extend-node-type.js]: You need to check if the user wants the pathPrefix used e.g. `gatsby build --prefix-paths`. If not, it should build normally without the path prefixing.

**Comment by KyleAMathews** [comment]
Linting failed https://travis-ci.org/gatsbyjs/gatsby/jobs/300020335

To fix, just run `npm run format` from the base of the repo.

File: packages/gatsby-transformer-remark/src/extend-node-type.js (hunk 1 above)
Comment (Swizec) [reply packages/gatsby-transformer-remark/src/extend-node-type.js]: Oooh I see, I have to use `pathPrefix` directly from the given argument, not go through the store. Thanks!

**Comment by Swizec** [comment]
Whoa @KyleAMathews what did you just do to make checks pass on my PR?

**Comment by KyleAMathews** [comment]
There were some failing tests on master that I fixed :-)

**Comment by KyleAMathews** [comment]
Merging! Thanks for the nice PR!

**Comment by jlengstorf** [comment]
Hiya @Swizec! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**